From a21db5fbba13c5bdf693a9ea4e4253cc41fcc1ba Mon Sep 17 00:00:00 2001
From: Matthew Waters <matthew@centricular.com>
Date: Fri, 3 Aug 2018 12:08:31 +1000
Subject: [PATCH] build: fix GMainLoop on macos for meson

Meson only takes the filename extension into account when choosing what
compiler to run so we need to copy the gmain.c which now contains objc
files and rename it to .m
---
 glib/meson.build | 13 +++++++++++--
 1 file changed, 11 insertions(+), 2 deletions(-)

diff --git a/glib/meson.build b/glib/meson.build
index ea9bf41..9f3e087 100644
--- a/glib/meson.build
+++ b/glib/meson.build
@@ -145,7 +145,7 @@ glib_sources = files(
   'glib-init.c',
   'glib-private.c',
   'glist.c',
-  'gmain.c',
+#  'gmain.c',
   'gmappedfile.c',
   'gmarkup.c',
   'gmem.c',
@@ -196,6 +196,14 @@ glib_sources = files(
   'gprintf.c',
 )
 
+if host_system == 'darwin'
+  glib_sources += configure_file(input : 'gmain.c',
+                                output : 'gmain.m',
+                                  copy : true)
+else
+  glib_sources += 'gmain.c'
+endif
+
 if host_system == 'windows'
   glib_win_rc = configure_file(
     input: 'glib.rc.in',
@@ -238,7 +246,8 @@ libglib = library('glib-2.0',
   include_directories : configinc,
   link_with : [charset_lib, gnulib_lib],
   dependencies : [pcre, thread_dep, libintl, librt] + libiconv + platform_deps,
-  c_args : ['-DG_LOG_DOMAIN="GLib"', '-DGLIB_COMPILATION'] + pcre_static_args + glib_hidden_visibility_args
+  c_args : ['-DG_LOG_DOMAIN="GLib"', '-DGLIB_COMPILATION'] + pcre_static_args + glib_hidden_visibility_args,
+  objc_args : ['-DG_LOG_DOMAIN="GLib"', '-DGLIB_COMPILATION'] + pcre_static_args + glib_hidden_visibility_args
 )
 
 libglib_dep = declare_dependency(
-- 
2.14.3 (Apple Git-98)

